Mage.Sets/src/mage/cards/p/ProgenitorsIcon.java

```diff
@@ -0,0 +1,127 @@
+import mage.watchers.Watcher;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class ProgenitorsIcon extends CardImpl {
+
+    public ProgenitorsIcon(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{3}");
+        
+
+        // As Progenitor's Icon enters the battlefield, choose a creature type.
+        // Based on Circle of Solace
+        this.addAbility(new AsEntersBattlefieldAbility(new ChooseCreatureTypeEffect(Outcome.Neutral)));
+
+        // {T}: Add one mana of any color.
+        this.addAbility(new AnyColorManaAbility());
+
+        // {T}: The next spell of the chosen type you cast this turn can be cast as though it had flash.
+        // Based on Ride the Avalanche
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new ProgenitorsIconAsThoughEffect(), new TapSourceCost());
+        ability.addWatcher(new ProgenitorsIconWatcher());
+        this.addAbility(ability);
+    }
+
+    private ProgenitorsIcon(final ProgenitorsIcon card) {
+        super(card);
+    }
+
+    @Override
+    public ProgenitorsIcon copy() {
+        return new ProgenitorsIcon(this);
+    }
+}
+
+class ProgenitorsIconAsThoughEffect extends AsThoughEffectImpl {
+
+    public ProgenitorsIconAsThoughEffect() {
+        super(AsThoughEffectType.CAST_AS_INSTANT, Duration.EndOfTurn, Outcome.Benefit);
+        staticText = "The next spell of the chosen type you cast this turn can be cast as though it had flash";
+    }
+
+    private ProgenitorsIconAsThoughEffect(final ProgenitorsIconAsThoughEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public void init(Ability source, Game game) {
+        ProgenitorsIconWatcher.addPlayer(source.getControllerId(), game);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public ProgenitorsIconAsThoughEffect copy() {
+        return new ProgenitorsIconAsThoughEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        if (!ProgenitorsIconWatcher.checkPlayer(affectedControllerId, game)) {
+            discard();
+            return false;
+        }
+        Card card = game.getCard(sourceId);
+        if (card == null) return false;
+
+        SubType subType = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game);
+        return card.getSubtype().contains(subType);
+    }
+}
+
+class ProgenitorsIconWatcher extends Watcher {
+
+    private final Set<UUID> playerSet = new HashSet<>();
+
+    public ProgenitorsIconWatcher() {
+        super(WatcherScope.GAME);
+    }
+
+    @Override
+    public void watch(GameEvent event, Game game) {
+        if (event.getType() == GameEvent.EventType.SPELL_CAST) {
+            playerSet.remove(event.getPlayerId());
```

Comment:
this isn't correct here, needs to only remove if the spell that was cast has the chosen type. might need some refactoring to handle correctly (e.g. instead of having a Set of player id, could have a Map of player id to subtype?)

Reply:
Used an Arraylist to handle instances where multiple instances of the card might choose the same or different subtypes for the same player